src/service/resources-impl.js

```diff
@@ -362,11 +362,21 @@ export class Resources {
       this.viewport_.ensureReadyForElements();
     }
 
-    // Create and add the resource.
-    const resource = new Resource((++this.resourceIdCounter_), element, this);
+    // First check if the resource is being reparented and if it requires
+    // reconstruction. Only already built elements are eligible.
+    let resource = Resource.forElementOptional(element);
+    if (resource &&
+        resource.getState() != ResourceState.NOT_BUILT &&
```

Comment:
If the element has already been built, we'll end up creating a new Resource for it. Should this be inside another conditional?

```js
if (resource && !reconstructWhenReparented) {
  if (!built) {
    //...
  }
} else {
  // new Resource
}